packages/playwright-test/src/store.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright Microsoft Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import type { TestStore } from '../types/test';
+import { currentConfig } from './common/globals';
+
+class JsonStore implements TestStore {
+  async get<T>(name: string) {
+    const file = this.path(name);
+    try {
+      const data = await fs.promises.readFile(file, 'utf-8');
+      return JSON.parse(data) as T;
+    } catch (e) {
+      return undefined;
+    }
+  }
+
+  path(name: string): string {
+    const config = currentConfig();
+    if (!config)
+      throw new Error('Cannot access store before config is loaded');
+    return path.join(config._internal.storeDir, name);
+  }
+
+  async set<T>(name: string, value: T | undefined) {
+    const file = this.path(name);
+    if (value === undefined) {
+      await fs.promises.rm(file, { force: true });
```

Comment:
Should we support `name` that is actually a directory? I'd say that would be convenient.

Reply:
Don't know, although convenient it maybe confusing.